mono/metadata/w32file-win32.c

```diff
@@ -235,33 +282,25 @@ guint32
 mono_w32file_get_attributes (const gunichar2 *name)
 {
 	guint32 res;
-	guint32 error;
 	HANDLE find_handle;
 	WIN32_FIND_DATA find_data;
 
 	MONO_ENTER_GC_SAFE;
-	res = GetFileAttributes (name);
-	error = GetLastError ();
-	MONO_EXIT_GC_SAFE;
-
-	if (res != INVALID_FILE_ATTRIBUTES)
-		return res;
 
-	if (error != ERROR_SHARING_VIOLATION)
-		return INVALID_FILE_ATTRIBUTES;
-
-	MONO_ENTER_GC_SAFE;
-	find_handle = FindFirstFile (name, &find_data);
-	MONO_EXIT_GC_SAFE;
-
-	if (find_handle == INVALID_HANDLE_VALUE)
-		return INVALID_FILE_ATTRIBUTES;
+	res = GetFileAttributes (name);
+	if (res == INVALID_FILE_ATTRIBUTES && GetLastError () == ERROR_SHARING_VIOLATION) {
+		find_handle = FindFirstFile (name, &find_data);
+		if (find_handle != INVALID_HANDLE_VALUE) {
```

Comment:
Sure, just copied pasted the original implementation.

Reply:
`WIN32_FIND_DATA`, `GetFileAttributes` etc.

There are multiple contradictory schools of thought here.

1. This a Win9x abstraction layer. Obsolete. Use the NT names directly.
2. These are the nice names. They aren't right because of the abstraction layer, but you can still use them.
3. Code should be as ifdef-independent as possible, so use the NT names directly.
4. Maybe Win9x/A isn't so dead. Very recently A can even mean UTF8. It almost seems good and we can now forget the entire wchar 30 year adventure (unless we write an actual text renderer).

Reply:
Agree, I started out using TCHAR together with Win32 in late 90's and got stuck to using the old API names since then. But since all our string parameters are currently explicit gunichar2 I agree that we should also be explicit in our API calls as well, using the W version of all methods instead of relying on compability macros.